src/entities/viewer/model/store.ts

```diff
@@ -0,0 +1,19 @@
+import { sample, createEvent, createStore, createEffect } from 'effector';
+import { User } from './types';
+import { getUser, login, logout, getAuthState } from './api';
+
+export const loginRequested = createEvent();
+export const logoutRequested = createEvent();
+
+export const loginFx = createEffect(login);
+export const logoutFx = createEffect(logout);
+export const getUserFx = createEffect(getUser);
+export const getAuthStateFx = createEffect(getAuthState);
+
+export const $user = createStore<User | null>(null).on(getUserFx.doneData, (_, user) => user);
```

Comment:
typescript should lint here that undefined is not compatible with type User | null

Reply:
Lint started saying it when I removed undefined from Promise<User | undefined>

Reply:
when i past <User | undefined> lint - does not object. When I wrote just getUser = (): Promise<User> => auth0.getUser(); Lint wrote about undefined. Why?